### PR TITLE
added webpush and apns-push configurations

### PR DESCRIPTION
**Pull request by andrewalf**
Working with this library I noticed that webpush configuration is not finished and can't be used. The same with apns-pushes. Tried to fix this, checked web notification modifications by setting an icon and it works.

Doc: https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages

### CODE REVIEW

**Comment by plokko** [comment]
Thanks, webpush and apns were added just as stubs for lack of time, i'll look into your code!
Thanks again for your contribution! 👍

**Comment by plokko** [comment]
I looked at your code, thanks for your submission and good work, very similar to my original code (glad you understood my work! )!
Only one question: what is `ApnsAlertConfig`? 
i can't find it in the docs!

**Comment by andrewalf** [comment]
Thanks for the quick response! I'm using this library in my work, and I really miss web and apns pushes :)
Generally, you wrote a good library, very close to fcm documentation, and saved lots of my time)

**Comment by andrewalf** [comment]
Sorry forgot phpdoc for the class. I'll add it in a minute

**Comment by plokko** [comment]
Shouldn't it be handled by `ApnsConfig`?
For what i've understood all this fields are handled by ApnsConfig headers so i'm more inclined to add helper functions like setTitle setLaunchImage that automatically set header values instead of creating a whole new class not present in official Firebase docs.
I'm trying to be as close as possible to official documentation (and Json-serialized result) to avoid confusion and because the main goal of the library is to be easly integrated in other wrappers.

**Comment by plokko** [comment]
No worry btw, you're my first contributor and a good one! :)
I did try to contribute to other firebase packages but all of them lacked something and they were not prone to updates so after a lot of struggle i decided to create my own from scratch, contributors are more than welcome!

File: src/FCM/Message/ApnsConfig.php [hunk 1]
```diff
@@ -1,8 +1,45 @@
...
 
+/**
+ * Class ApnsConfig
+ * @package Plokko\Firebase\FCM\Message
+ * @see https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#apnsconfig
+ 
+ * @property array $headers
+ * @property ApnsNotification $payload
+ */
 class ApnsConfig
```
Comment (plokko) [inline src/FCM/Message/ApnsConfig.php]: Shouldn't it implement implement JsonSerializable?!?

**Comment by andrewalf** [comment]
Honestly, a don't understand what do you mean saying "all these fields are handled by ApnsConfig headers". Headers are another property, not connected with stuff that is set through "alert" property.
For example, in webpush, the `WebPushNotification` class is also the representation of non-fcm entity: https://developer.mozilla.org/en-US/docs/Web/API/Notification

Also maybe you wanted to say that this should be handled by `ApnsNotification`, not `ApnsConfig`?
And in notification class add helpers like you said, or magic get/set. Smth like:
```php
class ApnsNotification implements JsonSerializable
{
  private $alert;

    function __get($k)
    {
        if ('title' === $k) {
          return $this->alert->title;
        }

        return $this->{$k};
    }

   function __set($k, $v)
    {
       if ('title' === $k) {
          $this->alert->title = $v;
          return;
       }
        $this->{$k}=$v;
    }
}
```

That's rather convinient, but dunno if it's neseccary. Anyway users will have to see the apple docs, to understand what properties mean.
What do you think?
By the way, i'll have some free time on weekends and I'll be glad to make this library even better. I could help you with refactoring and unit tests, if you don't mind

File: src/FCM/Message/ApnsConfig.php (hunk 1 above)
Comment (andrewalf) [reply src/FCM/Message/ApnsConfig.php]: Yes, it should. I'll fix it too)

**Comment by plokko** [comment]
@andrewalf no, you were correct, ApnsAlertConfig should be a subclass of ApnsConfig, i simply could not find it in the docs, i do approve your approach!
Yes DO add unit testing please!
I do need someone to write them because i have no free time to write them!

File: src/FCM/Message/ApnsConfig.php (hunk 1 above)
Comment (andrewalf) [reply src/FCM/Message/ApnsConfig.php]: Fixed this!

**Comment by andrewalf** [comment]
Thans man! Have a nice day :)

**Comment by andrewalf** [comment]
@plokko Don't forget about the new tag!)

**Pull request merged**